tests/std/tests/P0645R10_text_formatting_parsing/test.cpp

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <assert.h>
+#include <concepts>
+#include <format>
+#include <stdio.h>
+#include <string_view>
+
+using namespace std;
+
+// copied from the string_view tests
+template <typename CharT>
+struct choose_literal; // not defined
+
+template <>
+struct choose_literal<char> {
+    static constexpr const char* choose(const char* s, const wchar_t*) {
+        return s;
+    }
+};
+
+template <>
+struct choose_literal<wchar_t> {
+    static constexpr const wchar_t* choose(const char*, const wchar_t* s) {
+        return s;
+    }
+};
+
+#define TYPED_LITERAL(CharT, Literal) (choose_literal<CharT>::choose(Literal, L##Literal))
+
+template <typename _CharT>

```

Comment:
There are some unavoidable uses of `_Ugly` names here since we're testing internal machinery, but it would be nice to use pretty names consistently for non-internal-machinery so the distinction is clear.